qiita_pet/handlers/study_handlers/artifact.py

```diff
@@ -21,3 +24,47 @@ def get(self):
         artifact = to_int(self.get_argument('artifact_id'))
         self.write(artifact_graph_get_req(artifact, direction,
                                           self.current_user.id))
+
+
+class NewArtifactHandler(BaseHandler):
+    @authenticated
+    def get(self, study_id):
+        prep_files = [f for _, f in get_files_from_uploads_folders(study_id)
+                      if f.endswith(('txt', 'tsv'))]
+        artifact_types = artifact_types_get_req()['types']
+        data_types = sorted(data_types_get_req()['data_types'])
+        ontology = ena_ontology_get_req()
+        self.render("study_ajax/add_prep_artifact.html", prep_files=prep_files,
+                    artifact_types=artifact_types, data_types=data_types,
+                    ontology=ontology, study_id=study_id)
+
+    @authenticated
+    def post(self, study_id):
```

Comment:
It takes in the information for the prep template and artifact files, then creates the prep template and creates+attaches the initial artifact to the prep template.